src/_pytest/debugging.py

```diff
@@ -274,13 +275,8 @@ def pytest_pyfunc_call(self, pyfuncitem):
 def _test_pytest_function(pyfuncitem):
     _pdb = pytestPDB._init_pdb("runcall")
     testfunction = pyfuncitem.obj
-    pyfuncitem.obj = _pdb.runcall
-    if "func" in pyfuncitem._fixtureinfo.argnames:  # pragma: no branch
-        raise ValueError("--trace can't be used with a fixture named func!")
-    pyfuncitem.funcargs["func"] = testfunction
-    new_list = list(pyfuncitem._fixtureinfo.argnames)
-    new_list.append("func")
-    pyfuncitem._fixtureinfo.argnames = tuple(new_list)
+    pyfuncitem.obj = functools.partial(_pdb.runcall, testfunction)
+    pyfuncitem.obj.__name__ = testfunction.__name__
```

Comment:
that solution checks out proper,
nonetheless this feels like a hijacking of the pr to me

Reply:
> nonetheless this feels like a hijacking of the pr to me

￼why?
If it wasn't wanted then you could uncheck the box that allows it.
It can always be reset also (I've explicitly mentioned the old head).
I'm just trying to help fixing the issue (and limitations that were in place due to unknown reasons).

Why shouldn't I push hardening/improvements when I have them locally already, so that they can be reviewed/tested on CI?

@davidszotten 
What do you think?

Reply:
Reset it to  b421aef (from c45761c).  The test was timing out on CI, and I do not want to leave the impression to hijack / being hostile.
Will try to fix this in my fork, and then leave a comment here.

Reply:
the human interaction perception of such actions of merit can widely vary,
whiles some may be accepting of that, for me personally it would be the last time that i'd contribute to the project 

we should perhaps clarify on what we hope to accomplish by enabling maintainers to push commits to other prs

for me the main reason is to help new  contributors that struggle with the tooling,
for better solutions/code change suggestions outside of pre-commit,
i would always  strongly suggest to use something like proposed edits for guidance

for me there is a very important difference between guiding/mentoring and  accidentally taking over

and *i* perceive your actions as "taking over" instead of guiding/mentoring

Reply:
> for me personally it would be the last time that i'd contribute to the project

why?
It's not like anything has been merged here already, it's just still being worked on.  References are/were provided to revert/reset it, and IMHO it is far better to have CI results and an actual commit rather than a diff in a comment.  But I will stick to the latter then in the future here.

> strongly suggest to use something like proposed edits for guidance

I'm doing that also, but here it was not fitting.  Posting a diff then maybe instead.

Reply:
actions demonstrate values and projects that brush over contributors/new  contributors don't foster/respect them, and i don't want that kind of interaction

Reply:
agreed. though as feedback to you @RonnyPfannschmidt , of course @blueyed is also a contributor and maybe we can think of a better way to explain the behaviours we want to promote and why than something as brief as "this feels like a hijacking of the pr"